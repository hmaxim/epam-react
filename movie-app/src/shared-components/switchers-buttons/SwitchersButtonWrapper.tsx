import styled from "styled-components";

export interface ISwitcherButton {
    active: boolean;
    firstElement: boolean;
    lastElement: boolean;
}

const switcherButtonWrapper = styled.button`
  background: ${(props: ISwitcherButton) =>
    props.active ? "#F65261" : "rgba(85, 85, 85, 0.5)"};
  border-radius: 3px;
  border-top-left-radius: ${(props: ISwitcherButton) =>
    props.firstElement ? "3px" : "0px"};
  border-bottom-left-radius: ${(props: ISwitcherButton) =>
    props.firstElement ? "3px" : "0px"};
  border-bottom-right-radius: ${(props: ISwitcherButton) =>
    props.lastElement ? "3px" : "0px"};
  border-top-right-radius: ${(props: ISwitcherButton) =>
    props.lastElement ? "3px" : "0px"};
  color: white;
  outline: none;
  text-transform: uppercase;
  width: auto;
  padding: 5px 10px;
`;

export default switcherButtonWrapper;
