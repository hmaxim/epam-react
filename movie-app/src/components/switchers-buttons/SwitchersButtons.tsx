import React, { useState } from "react";
import SwitchersButtonWrapper from "./SwitchersButtonWrapper";

const SwitchersButtons = (props: any) => {
  const [buttons, updateSearchButtons] = useState(props.searchButtonsParams);

  const activateButton = (index: number) => {
    const newArr = [...buttons];
    newArr.forEach((item, i) =>
      i === index ? (item.active = true) : (item.active = false)
    );
    updateSearchButtons(newArr);
  };

  return (
    <div className="search-buttons-container">
      <span>{props.switchersTitle}</span>
      {buttons.map((button: any, index: number) => {
        return (
          <SwitchersButtonWrapper
            onClick={() => activateButton(index)}
            key={index}
            active={button.active}
            firstElement={index === 0}
            lastElement={index === buttons.length - 1}
          >
            {button.label}
          </SwitchersButtonWrapper>
        );
      })}
    </div>
  );
};

export default SwitchersButtons;
