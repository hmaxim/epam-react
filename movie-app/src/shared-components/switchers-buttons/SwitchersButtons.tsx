import React, { useState, useCallback, useEffect } from 'react';
import SwitchersButtonWrapper from './SwitchersButtonWrapper';

export interface ISwitchersButtons {
  searchButtonsParams: any[];
  click: (buttonValue: string) => void;
  switchersTitle: string;
  activeBtnValue?: string;
}

const SwitchersButtons = (props: ISwitchersButtons) => {
  const [buttons, updateSearchButtons] = useState(props.searchButtonsParams);

  const activateButton = useCallback(
    (index: number) => {
      const arr = buttons.map((item: any, i: number) => {
        i === index ? (item.active = true) : (item.active = false);
        return item;
      });

      if (props.click && props.activeBtnValue !== arr[index].buttonValue) {
        props.click(arr[index].buttonValue);
      }

      updateSearchButtons(arr);
    },
    [props.activeBtnValue],
  );

  return (
    <div className="search-buttons-container">
      <span>{props.switchersTitle}</span>
      {buttons.map((button: any, index: number) => {
        return (
          <SwitchersButtonWrapper
            onClick={() => {
              activateButton(index);
            }}
            key={button.label}
            id={button.label}
            active={button.active}
            firstElement={index === 0}
            lastElement={index === buttons.length - 1}
          >
            {button.label}
          </SwitchersButtonWrapper>
        );
      })}
    </div>
  );
};

export default SwitchersButtons;
