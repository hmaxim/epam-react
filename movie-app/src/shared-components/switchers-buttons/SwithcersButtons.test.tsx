import "jsdom-global/register";
import React from "react";
import { shallow, mount } from "enzyme";
import "jest-enzyme";
import "jest-styled-components";
import toJson from "enzyme-to-json";
import SwitchersButtons from "./SwitchersButtons";
import SwitchersButtonWrapper from "./SwitchersButtonWrapper";
import switcherButtonWrapper from "./SwitchersButtonWrapper";

const buttons = [
  {
    label: "test1",
    active: false
  },
  {
    label: "test2",
    active: true
  }
];

describe("SwitchersButtons component", () => {
  const component = mount(<SwitchersButtons searchButtonsParams={buttons} />);

  test("should call callBack and activate first button", () => {
    const notActiveButton = component.find("button");
    notActiveButton.first().simulate("click");
    expect(buttons[0].active).toBeTruthy();
  });

  test("should have props", () => {
    expect(component.props).toBeTruthy();
  });

  test("should define search-buttons-container", () => {
    expect(toJson(component.find(".search-buttons-container"))).toBeDefined();
  });

  test("should define search-buttons-container", () => {
    expect(toJson(component.find(SwitchersButtonWrapper))).toHaveLength(2);
  });
});
