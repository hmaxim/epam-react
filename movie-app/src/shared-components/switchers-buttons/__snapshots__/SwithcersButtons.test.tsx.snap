// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchersButtons component should be rendered 1`] = `
Array [
  .c0 {
  background: #F65261;
  border-radius: 3px;
  border-top-left-radius: 3px;
  border-bottom-left-radius: 3px;
  border-bottom-right-radius: 0px;
  border-top-right-radius: 0px;
  color: white;
  outline: none;
  text-transform: uppercase;
  width: auto;
  padding: 5px 10px;
}

<SwitchersButtonWrapper__switcherButtonWrapper
    active={true}
    firstElement={true}
    id="test1"
    key="test1"
    lastElement={false}
    onClick={[Function]}
  >
    <button
      className="c0"
      id="test1"
      onClick={[Function]}
    >
      test1
    </button>
  </SwitchersButtonWrapper__switcherButtonWrapper>,
  .c0 {
  background: rgba(85,85,85,0.5);
  border-radius: 3px;
  border-top-left-radius: 0px;
  border-bottom-left-radius: 0px;
  border-bottom-right-radius: 3px;
  border-top-right-radius: 3px;
  color: white;
  outline: none;
  text-transform: uppercase;
  width: auto;
  padding: 5px 10px;
}

<SwitchersButtonWrapper__switcherButtonWrapper
    active={false}
    firstElement={false}
    id="test2"
    key="test2"
    lastElement={true}
    onClick={[Function]}
  >
    <button
      className="c0"
      id="test2"
      onClick={[Function]}
    >
      test2
    </button>
  </SwitchersButtonWrapper__switcherButtonWrapper>,
]
`;
